reference/commands/lock/merge.rst

```diff
@@ -17,4 +17,130 @@ conan lock merge
+- It doesn't guarantee that the lockfile will be complete, might require ``--lockfile-partial`` if not
+- Recipe revisions, if defined, must contain the timestamp to be sorted correctly.
+
+
+.. warning::
+
+  - It is forbidden to manually manipulate a Conan lockfile, changing the strict sorting of references, and that could result in
+    any arbitrary undefined behavior.
+  - Recall that it is not possible to ``conan lock add`` a version range. The version might be not fully complete (like not providing
+    the revision), but it must be an exact version.
+
+
+This ``conan lock merge`` command can be useful to consolidate in a single lockfile when for some reasons there are several lockfiles
+that have diverged. A use case would be to create a multi-configuration lockfile that contains all necessary locked versions for
+all OSs (Linux, Windows, etc), even if there are conditional dependencies in the graph for the different OSs. At some point when
+testing a new dependency version, for example, ``pkg/3.4`` new version, when previously ``pkg/3.3`` was already in the graph, we
+might want to have such a new lockfile cleaning the previous ``pkg/3.3``. If we apply the ``--lockfile-clean`` argument that will
+remove the non-used versions in the lockfile, but that will also remove the OS-dependant dependencies. So something like this could be 
+done: lets say that we have this lockfile (simplified, removed revisions for simplicity) as the result of testing a new ``pkgb/0.2`` version
+for our main product ``app1/0.1``:
+
+.. code-block:: json
+  :caption: app.lock
+
+  {
+    "version": "0.5",
+    "requires": [
+        "pkgb/0.2",
+        "pkgb/0.1",
+        "pkgawin/0.1",
+        "pkganix/0.1",
+        "app1/0.1"
+    ]
+  }
+
+The ``pkgawin`` and ``pkganix`` are dependencies that exist exclusively in Windows and Linux respectively. Everything looks good,
+``pkgb/0.2`` new version works fine with our app, and we want to clean the unused things from the lockfile:
+
+.. code-block:: bash
+
+  $ conan lock create --requires=app1/0.1 --lockfile=app.lock --lockfile-out=win.lock -s os=Windows --lockfile-clean
+  # Note how both pkgb/0.1 and pkganix are gone
+  $ cat win.lock
+  {
+    "version": "0.5",
+    "requires": [
+        "pkgb/0.2",
+        "pkgawin/0.1",
+        "app1/0.1"
+    ]
+  }
+  $ conan lock create --requires=app1/0.1 --lockfile=app.lock --lockfile-out=nix.lock -s os=Linux --lockfile-clean
+  # Note how both pkgb/0.1 and pkgawin are gone
+  $ cat win.lock
+  {
+    "version": "0.5",
+    "requires": [
+        "pkgb/0.2",
+        "pkganix/0.1",
+        "app1/0.1"
+    ]
+  }
+    # Finally, merge the 2 clean lockfiles, for keeping just 1 for next iteration
+  $ conan lock merge --lockfile=win.lock --lockfile=nix.lock --lockfile-out=final.lock
+  $ cat final.lock
+  {
+    "version": "0.5",
+    "requires": [
+        "pkgb/0.2",
+        "pkgawin/0.1",
+        "pkganix/0.1",
+        "app1/0.1"
+    ]
+  }
```

Comment:
I would add a link to https://docs.conan.io/2/tutorial/versioning/lockfiles.html